### PR TITLE
Improve code style `Collection::sort()`

### PR DESCRIPTION
**Pull request by distantnative**
### Summary of changes
- Cleaned up `Collection::sort()` using some more modern PHP code, e.g `array_key_last()`

### Reasoning
- Moving our code base along bit by bit

## Ready?

- [x] In-code documentation (wherever needed)
- [x] Unit tests for fixed bug/feature
- [x] Tests and CI checks all pass

### CODE REVIEW

File: src/Toolkit/Collection.php [hunk 1]
```diff
@@ -961,49 +961,63 @@ public function sort(...$args): static
 		$fields = [];
 
 		foreach ($args as $arg) {
-			// get the index of the latest field array inside the $fields array
-			$currentField = $fields ? count($fields) - 1 : 0;
+			// get the index of the latest field array inside $fields
+			$field = array_key_last($fields);
```
Comment (bastianallgeier) [inline src/Toolkit/Collection.php:965]: Uhhh, I didn't know about this one. That's nice 💛

File: src/Toolkit/Collection.php [hunk 2]
```diff
@@ -961,49 +961,63 @@ public function sort(...$args): static
...
 
-			// detect the type of argument
-			// sorting direction
-			$argLower = is_string($arg) ? strtolower($arg) : null;
+			// normalize $arg
+			$arg = match (true) {
+				is_string($arg) => strtolower($arg),
+				default         => $arg
+			};
```
Comment (bastianallgeier) [inline src/Toolkit/Collection.php]: As much as I love the match clauses, I'm not sure I prefer it here over the old version to be honest. I found that line to be very readable. I would always use match when the optional lines get too long and complicated to read, but here it's really short and quite easy to scan.

File: src/Toolkit/Collection.php [hunk 3]
```diff
@@ -961,49 +961,63 @@ public function sort(...$args): static
...
+					'values' => A::map($array, function ($value) use ($collection, $arg) {
+						$value = $collection->getAttribute($value, $arg);
+
+						// make sure that we return something sortable
+						// but don't convert other scalars (especially numbers)
+						// to strings!
+						return match (true) {
+							is_scalar($value) => $value,
+							default           => (string)$value
+						};
```
Comment (bastianallgeier) [inline src/Toolkit/Collection.php]: It's the same for me here. I can definitely live with the match clause. I'm really just not sure if it is worth it to refactor such short lines.

File: src/Toolkit/Collection.php (hunk 2 above)
Comment (distantnative) [reply src/Toolkit/Collection.php]: For me personally it's easier to read logic vertically, I think. I always have troubles with the inline syntax to spot where is the condition where does the true-value start, where the false-value. But it's not a big issue. If you and/or others prefer it, I'll turn it back. 👍

Comment (bastianallgeier) [reply src/Toolkit/Collection.php]: I've just converted the three lines back.

**Review by bastianallgeier — APPROVED** [review]

**Pull request merged**